velox/vector/VectorBatch.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/buffer/Buffer.h"
+#include "velox/common/memory/ByteStream.h"
+#include "velox/common/memory/MappedMemory.h"
+#include "velox/common/memory/Memory.h"
+#include "velox/common/memory/StreamArena.h"
+#include "velox/type/Type.h"
+#include "velox/vector/SelectivityVector.h"
+
+namespace facebook::velox {
+
+class RowVector;
+
+namespace batch {
+
+enum class BatchSerdeStatus {
+  Success,
+  AllocationFailed,
+  SerializationError,
+  DeserializationError,
+  KeysSerializationFailed,
+  KeyIndexOutOfRange,
+  SizesDeserializationError
+};
+
+class VectorKeySerializer {
+ public:
+  /// Deserializes a set of keys in a vector an aggregated key
+  /// \param vector the input row vector
```

Comment:
For consistency's sake can we get rid of the \param's ; I have only seen it in one other place apart from UnsafeRowSerializer and while not explicitly mentioned I dont think it meets our coding guidelines.. (Keep the helpful descriptions on the parameters).